recipes/cpython/all/conanfile.py

```diff
@@ -0,0 +1,476 @@
+            del self.options.fPIC
+        if self.settings.compiler == "Visual Studio":
+            del self.options.lto
+            del self.options.docstrings
+            del self.options.pymalloc
+            del self.options.with_curses
+            del self.options.with_gdbm
+            del self.options.with_nis
+        if self._is_py2:
+            # Python 2.xx does not support following options
+            del self.options.with_lzma
+        elif self._is_py3:
+            # Python 3.xx does not support following options
+            del self.options.with_bsddb
+            del self.options.unicode
+
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if tools.is_apple_os(self.settings.os):
+            if self._is_py2:
+                # FIXME: python2 does not build on Macos due to a missing uuid_string_t type
+                raise ConanInvalidConfiguration("This recipe (currently) does not support building python2 for apple products.")
+        if self.settings.compiler == "Visual Studio":
+            # The msbuild generator only works with Visual Studio
+            self.generators.append("msbuild")
+            if not self.options.shared:
+                raise ConanInvalidConfiguration("MSVC does not support a static build")
+            if self._is_py2:
+                if self.settings.compiler.version >= tools.Version("14"):
+                    self.output.warn("Visual Studio versions 14 and higher are not supported")
+            if str(self.settings.arch) not in self._msvc_archs:
+                raise ConanInvalidConfiguration("Visual Studio does not support this architecure")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("Python-{}".format(self.version), self._source_subfolder)
+
+    def requirements(self):
+        self.requires("openssl/1.1.1g")
+        if not (self.settings.compiler == "Visual Studio" and tools.Version(self.version) >= tools.Version("3.8")):
+            self.requires("expat/2.2.9")
+        self.requires("mpdecimal/2.4.2")
+        self.requires("zlib/1.2.11")
+        if self.settings.compiler != "Visual Studio" or tools.Version(self.version) >= "3.8":
+            self.requires("libffi/3.3")
+        if self.settings.os != "Windows":
+            self.requires("libuuid/1.0.3")
+            self.requires("libxcrypt/4.4.16")
+        if self.options.with_bz2:
+            self.requires("bzip2/1.0.8")
+        if self.options.get_safe("with_gdbm", False):
+            self.requires("gdbm/1.18.1")
+        if self.options.get_safe("with_nis", False):
+            raise ConanInvalidConfiguration("nis is not available on CCI (yet)")
+            self.requires("nis/x.y.z")
+        if self.options.with_sqlite3:
+            self.requires("sqlite3/3.31.1")
+        if self.options.with_tkinter:
```

Comment:
```suggestion
        if self.options.with_tkinter:
          # TODO: Add tk when available
```

TODO messages can be easily filtered